File: client-sdk/csharp/sdk/src/VertexCache/Sdk/Results/VCacheResult.cs
namespace VertexCache.Sdk.Results
{
    public class VCacheResult
    {
        public bool IsSuccess { get; set; }
        public string Message { get; set; } = string.Empty;
        public VCacheErrorCode Code { get; set; } = VCacheErrorCode.None;
        public string? Raw { get; set; }

        public VCacheResult() { }

        public static VCacheResult SuccessWithRaw(string raw)
        {
            return new VCacheResult
            {
                IsSuccess = true,
                Message = raw,
                Code = VCacheErrorCode.None,
                Raw = raw
            };
        }

        public static VCacheResult FailureWithRaw(VCacheErrorCode code, string message, string raw)
        {
            return new VCacheResult
            {
                IsSuccess = false,
                Message = message,
                Code = code,
                Raw = raw
            };
        }

        public static VCacheResult Success(string message)
        {
            return new VCacheResult
            {
                IsSuccess = true,
                Message = message,
                Code = VCacheErrorCode.None
            };
        }

        public static VCacheResult Failure(VCacheErrorCode code, string message)
        {
            return new VCacheResult
            {
                IsSuccess = false,
                Message = message,
                Code = code
            };
        }

        public override string ToString()
        {
            return IsSuccess ? $"+{Message}" : $"-{Message}";
        }
    }
}

File: client-sdk/go/sdk/results/vcache_result.go
package results

import (
	"fmt"
)

type VCacheResult struct {
	IsSuccess bool
	Message   string
	Code      VCacheErrorCode
	Raw       *string
}

func NewSuccessWithRaw(raw string) *VCacheResult {
	return &VCacheResult{
		IsSuccess: true,
		Message:   raw,
		Code:      None,
		Raw:       &raw,
	}
}

func NewErrorWithRaw(code VCacheErrorCode, message string, raw string) *VCacheResult {
	return &VCacheResult{
		IsSuccess: false,
		Message:   message,
		Code:      code,
		Raw:       &raw,
	}
}

func NewNil() *VCacheResult {
	return &VCacheResult{
		IsSuccess: true,
		Message:   "(nil)",
		Code:      None,
		Raw:       nil,
	}
}

func Success(message string) *VCacheResult {
	return &VCacheResult{
		IsSuccess: true,
		Message:   message,
		Code:      None,
	}
}

func Failure(code VCacheErrorCode, message string) *VCacheResult {
	return &VCacheResult{
		IsSuccess: false,
		Message:   message,
		Code:      code,
	}
}

func (r *VCacheResult) String() string {
	if r.IsSuccess {
		return fmt.Sprintf("+%s", r.Message)
	}
	return fmt.Sprintf("-%s", r.Message)
}
